bin/setup-popsyn

```diff
@@ -0,0 +1,180 @@
+from posydon.utils.common_functions import convert_metallicity_to_string
+import argparse
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('metallicity', type=float)
+    args = parser.parse_args()
+    ini_kw = binarypop_kwargs_from_ini("{ini_file}")
+    ini_kw["metallicity"] = args.metallicity
+    str_met = convert_metallicity_to_string(args.metallicity)
+    ini_kw["temp_directory"] = str_met+"_Zsun_" + ini_kw["temp_directory"]
+    synpop = BinaryPopulation(**ini_kw)
+    synpop.evolve()\n'''
+    filename =f'run_metallicity.py'
+    with open(filename, mode='w') as file:
+        file.write(text)
+
+def create_merge_script(ini_file):
+    
+    filename='merge_metallicity.py'
+    with open(filename, mode='w') as file:
+        file.write(f'''from posydon.popsyn.binarypopulation import BinaryPopulation
+from posydon.popsyn.io import binarypop_kwargs_from_ini
+from posydon.utils.common_functions import convert_metallicity_to_string
+import argparse
+import os
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('metallicity', type=float)
+    args = parser.parse_args()
+    ini_kw = binarypop_kwargs_from_ini("{ini_file}")
+    ini_kw['metallicity'] = args.metallicity
+    str_met = convert_metallicity_to_string(args.metallicity)
+    ini_kw['temp_directory'] = str_met+"_Zsun_" + ini_kw["temp_directory"]
+    synpop = BinaryPopulation(**ini_kw)
+    path_to_batch = ini_kw['temp_directory']
+    tmp_files = [os.path.join(path_to_batch, f)    \\
+                         for f in os.listdir(path_to_batch) \\
+                            if os.path.isfile(os.path.join(path_to_batch, f))]
+    tmp_files = sorted(tmp_files, key=lambda x: int(x.split(".")[-1]))
+    synpop.combine_saved_files(str_met+ '_Zsun_population.h5', tmp_files)
+    print('done')
+
+    if len(os.listdir(path_to_batch)) == 0:
+        os.rmdir(path_to_batch)
+
+''')
+
+def create_slurm_submit(metallicity, job_array_length, partition, email, walltime, account, path_to_posydon, path_to_posydon_data):
+    '''creates the slurm submit script'''
+
+    str_met = convert_metallicity_to_string(metallicity)
+    job_array_length = job_array_length-1
+
+    text = f'''#!/bin/bash
+#SBATCH --array=0-{job_array_length}
+#SBATCH --job-name=pop_syn_{str_met}
+#SBATCH --output=./{str_met}_logs/popsyn_{str_met}_%A_%a.out
+#SBATCH --time={walltime}
+#SBATCH --mem-per-cpu=4G
+'''
+    if account != None:
+        text += f'#SBATCH --account={account}\n'
+    if partition!=None:
+        text += f'#SBATCH --partition={partition}\n'
+    if email != None:
+        text = text + \
+    f'''#SBATCH --mail-type=FAIL
+#SBATCH --mail-user={email}
+'''
+
+    text = text + \
+    f'''export PATH_TO_POSYDON={path_to_posydon}
+export PATH_TO_POSYDON_DATA={path_to_posydon_data}
+python ./run_metallicity.py {metallicity}
+'''
+    filename = f"slurm_array_{str_met}_Zsun.sl"
```

Comment:
Here we should decide on a convention how to call slurm scripts. This is now the third way:

- in the girds the files are named `slurm*.sh`
- in the pipeline the files are names `*.slurm`
- and now in the pop-synth they get called `*.sl`